repronim/environment/dockerenvironment.py

```diff
@@ -0,0 +1,165 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the repronim package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Environment sub-class to provide management of environment engine."""
+
+from io import BytesIO
+from ..support.exceptions import CommandError
+
+from repronim.environment.base import Environment
+
+
+class DockerEnvironment(Environment):
+    """
+    Environment manager which talks to a Docker engine.
+    """
+
+    def __init__(self, config = {}):
+        """
+        Class constructor
+
+        Parameters
+        ----------
+        config : dictionary
+            Configuration parameters for the environment.
+        """
+        if not 'base_image_id' in config:
```

Comment:
in general `x not in` is preferred to `not x in` (pycharm should have hinted on that I believe)